### PR TITLE
Bulk load service

### PR DESCRIPTION
**Pull request by jSylvestre**
Mostly tested

### CODE REVIEW

**Comment by jSylvestre** [comment]
Maybe load all the sponsor accounts first....

**Comment by jSylvestre** [comment]
Issue #145

File: Hippo.Web/Services/BulkLoadService.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                    Log.Error($"Kerb not found: {kerb}");
+                    continue;
+                }
+                if(!await _dbContext.Users.AnyAsync(a => a.Iam == user.Iam))
+                {
+                    Log.Information($"Adding user: {kerb}");
+                    await _dbContext.Users.AddAsync(user);
+                }
+                kerbsInDb.Add(kerb);
+            }
```
Comment (sprucely) [inline Hippo.Web/Services/BulkLoadService.cs:63]: Not sure if multiple calls to identity service can be avoided, but multiple calls to db certainly can without getting fancy with udf/temp table stuff if you load all the kerbs into memory (swap Iam for Id if you want to search by that) 
```csharp
var currentKerbs = _dbContext.Users.Select(x => x.Id).ToHashSet();
var kerbsToBeAdded = kerbs.Where(id => !currentKerbs .Contains(id));
```
I wouldn't expect the number of user records to ever get high enough to cause a memory issue.

File: Hippo.Web/Services/BulkLoadService.cs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        public async Task<int> Load(Cluster cluster, string data)
+        {
+            var count = 0;
+            var pairs = data.Split(",");
+
+            Log.Information($"Starting bulk load for {pairs.Count()} accounts");
+
+            //Create missing users
+            var kerbs = new List<string>();
+            var kerbsInDb = new List<string>();
```
Comment (sprucely) [inline Hippo.Web/Services/BulkLoadService.cs:36]: This should probably be a `HashSet<string>` if you're going to be calling `.Contains(...)` on it in a loop.

**Review by sprucely — APPROVED** [review]
Haven't tested anything, but it looks okay to me. Left a couple comments about things that can be done to improve performance.

**Comment by jSylvestre** [comment]
Closing without merging as this wouldn't work with the new way accounts will be done.

**Pull request closed without merging**